instagram/src/components/Login/Login.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import './Login.css';
+
+import AppsImg from '../../images/apps.png';
+import InstagramLogo from '../../images/instagram-name-logo.jpg';
+import PhonesImg from '../../images/phones.jpg';
+
+class Login extends Component {
+    state = {
+        username: ""
+    }
+
+    handleChanges = e => {
+        this.setState({
+            [e.target.name]: e.target.value
+        })
+    }
+
+    handleLogin = e => {
+        //prevent default behavior of <form> element
+        e.preventDefault();
+
+        //check that the user input a username
+        if(!this.state.username) alert('Username field cannot be empty'); 
```

Comment:
good accounting for the edge case! I personally also like writing customer alert messages for such events, but also remember that you can add the `required` attribute to the form input to do the same thing with less code (but unfortunately, only with a default message)